interface/Tree.h

```diff
@@ -15,30 +15,29 @@
 class Tree {
 
   public:
-    Tree(Rectangle, int);
+    Tree(const Rectangle, int);
     ~Tree();
 
     bool empty();
     void addCell(const Cell*);
     int countCells();
-    std::vector<const Cell*>* getCells();
-    Tree* getLeaf(Point);
+    std::unique_ptr<std::vector<const Cell*>> getCells();
```

Comment:
Wouldn't it be simpler to return `const std::vector<const Cell*>&` and define `cells_` as `std::vector<const Cell*> cells_` ( no pointer).
In your case you have to move the `unique_ptr` which means that `cells_` is not valid anymore.